File: server/schemas/student.py
from pydantic import BaseModel
from typing import Optional
from datetime import date, time, timedelta
from datetime import datetime


class BookAppointmentRequest(BaseModel):
    faculty_id: int
    date: date
    start_time: time
    end_time: time
    purpose: str
    description: str

class StudentStats(BaseModel):
    pending: int
    confirmed: int
    completed: int

class StudentProfileUpdate(BaseModel):
    name: Optional[str] = None
    phone: Optional[str] = None
    semester: Optional[int] = None

File: client/src/types/student.ts
export interface Faculty {
  user_id: number;
  name: string;
  designation: string;
  office: string;
  research_interests: string[]; // Our new backend sends this as a list
  department_name: string;
  busy: boolean;
}

export interface Appointment {
  id: number;
  faculty_id: number;
  faculty_name: string;
  date: string; // ISO format
  time: string;
  purpose: string;
  description: string;
  status: 'pending' | 'confirmed' | 'declined' | 'cancelled';
}

export interface StudentStats {
  pending: number;
  confirmed: number;
  completed: number;
}
